packages/core/src/Asset/NftMetadata/fromMetadatum.ts

```diff
@@ -96,39 +96,57 @@ const getAssetMetadata = (policy: Cardano.MetadatumMap, assetName: Cardano.Asset
       })()
   );
 
+type AssetIdParts = Pick<AssetInfo, 'policyId' | 'name'>;
+const getName = (assetMetadata: Cardano.MetadatumMap, version: string, asset: AssetIdParts, logger: Logger) => {
+  const name = asString(assetMetadata.get('name'));
+  if (name) return name;
+  if (version === '1.0') {
+    try {
+      return AssetName.toUTF8(asset.name);
+    } catch (error) {
+      logger.warn(error);
+    }
+  }
+};
+
+const parseVersion = (version: Cardano.Metadatum | undefined) => {
+  if (!version) return '1.0';
+  if (typeof version === 'bigint') {
+    return `${version}.0`;
+  }
+  const stringVersion = asString(version);
+  if (stringVersion) {
+    return `${Number(stringVersion)}.0`;
```

Comment:
I don't know which is the source of this data, so how can't tell how much my suggestion is valid or could be ignored.

Here may worth to validate the string (something like a RegExp to ensure it contains only digits...)

Reply:
Nice catch @iccicci ! Indeed it can be any string, added some validation 78adbbb556aa8dec8ac317d5ff3e431829830c8a